internal/controller/config.go

```diff
@@ -22,67 +22,122 @@ import (
 
 const (
 	metricsAddrFlagName                = "metrics-addr"
+	metricsBindAddressFlagName         = "metrics-bind-address"
+	metricsPortFlagName                = "metrics-port"
+	webhookServerBindAddressFlagName   = "webhook-server-bind-address"
+	webhookServerPortFlagName          = "webhook-server-port"
+	webhookServerTLSServerCertDir      = "webhook-server-tls-server-cert-dir"
 	enableLeaderElectionFlagName       = "enable-leader-election"
 	leaderElectionIDFlagName           = "leader-election-id"
 	leaderElectionResourceLockFlagName = "leader-election-resource-lock"
-	ignoreOperationAnnotationFlagName  = "ignore-operation-annotation"
 	disableLeaseCacheFlagName          = "disable-lease-cache"
 
-	defaultMetricsAddr                = ":8080"
+	defaultMetricsAddr                = ""
+	defaultMetricsBindAddress         = ""
+	defaultMetricsPort                = 8080
+	defaultWebhookServerBindAddress   = ""
+	defaultWebhookServerPort          = 9443
+	defaultWebhookServerTLSServerCert = "/etc/webhook-server-tls"
 	defaultEnableLeaderElection       = false
 	defaultLeaderElectionID           = "druid-leader-election"
 	defaultLeaderElectionResourceLock = resourcelock.LeasesResourceLock
-	defaultIgnoreOperationAnnotation  = false
 	defaultDisableLeaseCache          = false
 )
 
+// ServerConfig contains details for the HTTP(S) servers.
+type ServerConfig struct {
+	// Webhook is the configuration for the HTTPS webhook server.
+	Webhook HTTPSServer
+	// Metrics is the configuration for serving the metrics endpoint.
+	Metrics *Server
+}
+
+// HTTPSServer is the configuration for the HTTPSServer server.
+type HTTPSServer struct {
+	// Server is the configuration for the bind address and the port.
+	Server
+	// TLSServer contains information about the TLS configuration for an HTTPS server.
+	TLS TLSServer
+}
+
+// TLSServer contains information about the TLS configuration for an HTTPS server.
+type TLSServer struct {
+	// ServerCertDir is the path to a directory containing the server's TLS certificate and key (the files must be
+	// named tls.crt and tls.key respectively).
+	ServerCertDir string
+}
+
+// Server contains information for HTTP(S) server configuration.
+type Server struct {
+	// BindAddress is the IP address on which to listen for the specified port.
+	BindAddress string
+	// Port is the port on which to serve unsecured, unauthenticated access.
+	Port int
```

Comment:
Ports in IPv4 are unsigned 16 bit integers. This should be made `uint16`.
Validation of the input can be delegated to the pflag package if this is done.

Reply:
Thanks for pointing this out. This was existing previously - in this PR we have just moved it so did not pay much attention to this. I have changed this now.

Reply:
It's surprising that there is no value validation done for ports in controller-runtime.
But yeah, we can just stay consistent with `webhook.Options` for now.